lib/factory.ts

```diff
@@ -124,10 +124,12 @@ export class Factory<T, I = any> {
    * @param params
    * @returns a new factory
    */
-  params(params: DeepPartial<T>): this {
+  params<TOverride extends T = T>(
+    params: DeepPartial<TOverride>,
+  ): Factory<TOverride, I> {
     const factory = this.clone();
     factory._params = merge({}, this._params, params, mergeCustomizer);
-    return factory;
+    return factory as unknown as Factory<TOverride, I>;
```

Comment:
The issue, and I'm not sure yet how to solve it, is that even though `TOverride` extends `T`, we aren't guaranteeing that new params are passed in such that the built object is actually a `TOverride`.

Reply:
You are absolutely right, `params` is `Partial` and might not contain all properties required for `TOverride`. 🤔 I also can’t think about any other approach to make it work with the `params()` method. Maybe with a dedicated `extends()` method that accepts another generator function.